lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1155,6 +1155,53 @@ findFieldWithName(const std::vector<swift::reflection::FieldInfo> &fields,
   return child_indexes.size();
 }
 
+static llvm::Optional<std::string>
+GetMultiPayloadEnumCaseName(const swift::reflection::EnumTypeInfo *eti,
+                            const DataExtractor &data) {
+  using namespace swift::reflection;
+  assert(eti->getEnumKind() == EnumKind::MultiPayloadEnum);
+  const auto &cases = eti->getCases();
+  auto it = std::max_element(cases.begin(), cases.end(),
+                             [](const auto &a, const auto &b) {
+                               return a.TI.getSize() < b.TI.getSize();
+                             });
+  if (it == cases.end())
+    return {};
+
+  auto payload_capacity = it->TI.getSize();
+  if (data.GetByteSize() == payload_capacity + 1) {
+    auto tag = data.GetDataStart()[payload_capacity];
+    if (tag >= 0 && tag < cases.size())
+      return cases[tag].Name;
+  }
+
+  return {};
+}
```

Comment:
@adrian-prantl @tbkka this is the manual implementation of projecting multipayload enums, where just the case name is taken.

Reply:
This looks like the right idea.  As long as you use this sparingly (ideally, only for certain well-known types), I would call it a reasonable workaround until the better solution I've been chipping away at is complete.

In particular, as you and I have discussed, the compiler uses a variety of layout strategies; this only supports the simplest one.  The "better solution" I've been working on is taking a while precisely because I need to know which strategy the compiler used and accurately support each one.